File: 866-rectangle-overlap/rectangle-overlap.go
func isRectangleOverlap(rec1 []int, rec2 []int) bool {
    // Check overlap on X-axis
    xOverlap := rec1[0] < rec2[2] && rec2[0] < rec1[2]

    // Check overlap on Y-axis
    yOverlap := rec1[1] < rec2[3] && rec2[1] < rec1[3]

    return xOverlap && yOverlap
}

File: 866-rectangle-overlap/rectangle-overlap.py
class Solution:
    def isRectangleOverlap(self, rec1: list[int], rec2: list[int]) -> bool:
        return (
            max(rec1[0], rec2[0]) < min(rec1[2], rec2[2])
            and
            max(rec1[1], rec2[1]) < min(rec1[3], rec2[3])
        )